hpccm/templates/ConfigureCMake.py

```diff
@@ -0,0 +1,139 @@
+
+# Copyright (c) 2018, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=invalid-name, too-few-public-methods
+
+"""Documentation TBD"""
+
+from __future__ import absolute_import
+from __future__ import unicode_literals
+from __future__ import print_function
+from six.moves import shlex_quote
+
+import logging  # pylint: disable=unused-import
+import os
+
+
+class ConfigureCMake(object):
+    """Documentation TBD"""
+
+    def __init__(self, **kwargs):
+        """Documentation TBD"""
+
+        #super(ConfigureMake, self).__init__()
+        self.__build_directory = None
+
+        # Some components complain if some compiler variables are
+        # enabled, e.g., MVAPICH2 with F90, so provide a way for the
+        # caller to disable any of the compiler variables.
+        self.toolchain_control = kwargs.get('toolchain_control',
+                                            {'CC': True, 'CXX': True,
+                                             'F77': True, 'F90': True,
+                                             'FC': True})
+
+    def build_step(self, target='all', parallel=4):
+        """Documentation TBD"""
+        return 'cmake --build {0} --target {1} -- -j{2:d}'.format(
+            self.__build_directory, target, parallel)
+
+    def configure_step(self, directory=None, build_directory='build',
```

Comment:
I'm not a cmake expert, but I think they refer to this part of the workflow as `generate`?  Rename `configure_step` to `generate_step`?